src/ucp/core/ucp_worker.c

```diff
@@ -246,17 +246,17 @@ uct_to_ucp_error_proxy(void *arg, uct_ep_h uct_ep, ucs_status_t status)
     return;
 
 found_ucp_ep:
-    ucp_ep_config(ucp_ep)->status = status;
-
-/* Destroy all lanes excepting failed one since ucp_ep becomes unusable as well */
+    /* Destroy all lanes excepting failed one since ucp_ep becomes unusable as well */
     for (lane = 0, n_lanes = ucp_ep_num_lanes(ucp_ep); lane < n_lanes; ++lane) {
         if ((lane != failed_lane) && ucp_ep->uct_eps[lane]) {
             uct_ep_destroy(ucp_ep->uct_eps[lane]);
             ucp_ep->uct_eps[lane] = NULL;
-            ucp_ep_config(ucp_ep)->key.num_lanes--;
         }
     }
-    ucs_assert(ucp_ep_config(ucp_ep)->key.num_lanes == 1);
+
+    ucp_ep_config_key_t key = ucp_ep_config(ucp_ep)->key;
+    ucp_ep->cfg_index = ucp_worker_get_ep_config(worker, &key);
```

Comment:
this is not right - need to fill the key **before** calling this function, otherwise it's pointless